ProjectBasic/app/src/main/java/com/example/moviedb/screen/home/HomeFragment.kt

```diff
@@ -12,13 +16,55 @@ class HomeFragment : BaseFragment<HomeFragmentBinding, HomeViewModel>() {
     override val layoutId: Int
         get() = R.layout.home_fragment
     override val viewModel: HomeViewModel by viewModel()
+    private var genresSelected = 0
 
     override fun initComponents(view: HomeFragmentBinding) {
+        viewModel.eventNetworkError.observe(
+            viewLifecycleOwner,
+            Observer<Boolean> { isNetworkError ->
+                if (isNetworkError) onNetworkError()
+            })
+
+        viewModel.listGenres.observe(viewLifecycleOwner, Observer { genres ->
+            if (movieByKeyTabLayout?.tabCount == 0) {
+                activity?.let {
+                    movieByKeyTabLayout?.setTabTextColors(
+                        ContextCompat.getColor(it, R.color.colorTextLightBlue),
+                        ContextCompat.getColor(it, R.color.colorOrange)
+                    )
+                }
+                for (element in genres) {
+                    movieByKeyTabLayout.addTab(
+                        movieByKeyTabLayout.newTab().setText(element.genresName)
+                    )
+                }
+                genresSelected = genres.getOrNull(0)?.genresID ?: 0
+            }
+        })
+        movieByKeyTabLayout?.addOnTabSelectedListener(
```

Comment:
được thì nên implement sẽ đẹp code hơn nhé chú
```suggestion

        movieByKeyTabLayout?.addOnTabSelectedListener(
```